worker.go

```diff
@@ -93,12 +102,13 @@ func startWorkers(fileName string, nbWorkers int, env PreparedEnv) error {
 
 				// TODO: make the max restart configurable
 				if _, ok := workersRequestChans.Load(absFileName); ok {
-					workersReadyWG.Add(1)
 					if fc.exitStatus == 0 {
 						if c := l.Check(zapcore.InfoLevel, "restarting"); c != nil {
 							c.Write(zap.String("worker", absFileName))
 						}
 					} else {
+						// we will wait a few milliseconds to not overwhelm the logger in case of repeated unexpected terminations
+						time.Sleep(50 * time.Millisecond)
```

Comment:
This was a problem in the case where there was an error in the worker script itself (while it's being watched). I think it should keep retrying, but not waiting a few ms here will make my console get stuck for several minutes because of the sheer amount of error messages (multiple every ms).

I think this is definitely necessary, but maybe there's a better workaround?

Reply:
Check out #1038

Reply:
An exponential backoff is probably a better approach 👍 . I'll wait until your branch is merged